paasta_tools/instance/kubernetes.py

```diff
@@ -73,6 +75,44 @@ def set_cr_desired_state(
         raise RuntimeError(error_message)
 
 
+def autoscaling_status(
+    kube_client: kubernetes_tools.KubeClient,
+    job_config: LongRunningServiceConfig,
+    namespace: str,
+):
+    status = {}
+    try:
+        hpa = kube_client.autoscaling.read_namespaced_horizontal_pod_autoscaler(
+            name=job_config.get_sanitised_deployment_name(), namespace=namespace
+        )
+        status["min_instances"] = hpa.spec.min_replicas
+        status["max_instances"] = hpa.spec.max_replicas
+        # Parse metrics sources, based on
+        # https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V2beta1ExternalMetricSource.md#v2beta1externalmetricsource
+        metric_stats = []
+        parser = HPAMetricsParser(hpa)
+        if hpa.spec.metrics is not None:
+            for metric_spec in hpa.spec.metrics:
+                metric_stats.append(parser.parse_target(metric_spec))
+        if hpa.status.current_metrics is not None:
+            for metric_spec in hpa.status.current_metrics:
+                metric_stats.append(parser.parse_current(metric_spec))
+        status["metrics"] = metric_stats
+        status["desired_replicas"] = hpa.status.desired_replicas
+        status["last_scale_time"] = (
+            hpa.status.last_scale_time.replace(tzinfo=pytz.UTC).isoformat()
+            if getattr(hpa.status, "last_scale_time")
+            else "N/A"
+        )
+    except ApiException as e:
+        if e.status == 404:
+            # This means that HPA doesn't exist. Return nothing.
+            return {}
```

Comment:
Out of curiosity, will this mask the error that HPA doesn't exist but it should?

Reply:
Good point.